rust/worker/src/compactor/runtime.rs

```diff
@@ -0,0 +1,115 @@
+use std::sync::Arc;
+use std::sync::RwLock;
+use std::thread;
+use std::time::Duration;
+use std::time::Instant;
+pub(crate) trait Runnable {
+    fn run(&self);
+    fn box_clone(&self) -> Box<dyn Runnable + Send + Sync>;
+}
+
+impl Clone for Box<dyn Runnable + Send + Sync> {
+    fn clone(&self) -> Box<dyn Runnable + Send + Sync> {
+        self.box_clone()
+    }
+}
+
+#[derive(Clone)]
+pub(crate) struct Runtime {
+    runable: Box<dyn Runnable + Send + Sync>,
+    running: Arc<RwLock<bool>>,
+    interval: Option<Duration>,
+}
+
+impl Runtime {
+    pub(crate) fn execute(&self) -> RuntimeHandle {
+        let rt = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+        let runnable = self.runable.box_clone();
+        let running = self.running.clone();
+        let interval = self.interval;
+        let join_handle = thread::spawn(move || {
```

Comment:
Is every compaction 1:1 with a thread - how would you run compaction on multiple threads, does the underlying code have to be responsible for spawning new threads / how is balancing achieved there?